### DISCUSSION TITLE
How to understand the routing table of otbr?

### DISCUSSION THREAD

**Discussion by luyou-tuya** [openthread/openthread#8649] [Q&A] [answered]
The routing table of otbr is as follows, and the information output by the routing table cannot be well understood
otbr RLOC16 is  0xd800 
If I want to send data to the sub-device whose RLOC is 0x1802, the data packet reaches otbr, otbr searches the routing table, finds that the next hop is router id 38, and then it will be sent to the route with route id 38 for forwarding, so understand ? Why doesn't it send directly to its parent node rloc 0x1800。
```
| ID | RLOC16 | Next Hop | Path Cost | LQ In | LQ Out | Age | Extended MAC     | Link |
+----+--------+----------+-----------+-------+--------+-----+------------------+------+
|  2 | 0x0800 |       28 |         1 |     3 |      3 |  27 | f67f590fadf06b9e |    1 |
|  6 | 0x1800 |       38 |         1 |     3 |      2 |  52 | 3622b3e959353775 |    1 |
|  9 | 0x2400 |       38 |         1 |     3 |      2 |  24 | fe4bf19586658d38 |    1 |
| 12 | 0x3000 |       28 |         1 |     3 |      3 |  22 | 6e169d7c2a72bf25 |    1 |
| 16 | 0x4000 |       38 |         1 |     3 |      3 |  20 | bed500d869ea9e5e |    1 |
| 18 | 0x4800 |       28 |         1 |     3 |      3 |  55 | e21217a2a4ebd131 |    1 |
| 21 | 0x5400 |       38 |         1 |     3 |      3 |  25 | 02529122593f24e3 |    1 |
| 23 | 0x5c00 |       28 |         1 |     3 |      3 |  26 | 72263fe00cf470b0 |    1 |
| 27 | 0x6c00 |       38 |         1 |     2 |      2 |  26 | 429cb9ef5cf19f0d |    1 |
| 28 | 0x7000 |       38 |         1 |     3 |      3 |  16 | 56db062927ff33b2 |    1 |
| 30 | 0x7800 |       28 |         1 |     3 |      3 |  11 | 82d41cb0d3fabfc2 |    1 |
| 38 | 0x9800 |       28 |         1 |     3 |      3 |  24 | 8696709b7db12cb9 |    1 |
| 46 | 0xb800 |       38 |         1 |     3 |      3 |  12 | 4eaad145844a8ff2 |    1 |
| 54 | 0xd800 |       63 |         0 |     0 |      0 |   0 | 060709fdda73ea45 |    0 |
```

**Comment by jwhui** [comment, marked as answer]
Next Hop represents the best path when routing through a neighboring router.

When forwarding a packet, the router will consider both a direct link and other through another router. In this case, the routing table shows a direct link to 0x1800 and will use that when forwarding the packet.

Having both options provides greater robustness to link failures.

**Comment by sohaibbinaltaf** [comment]
If you could please tell me which command can be used to get this routing table?

**Reply by abtink** [reply]
`router table`

You check the [`src/cli/README.md`](https://github.com/openthread/openthread/tree/main/src/cli#openthread-cli-reference) for all CLI commands 
(also documentation for the CLI is being added in https://openthread.io/reference/cli/commands).
